internal/acceptance/testcase.go

```diff
@@ -123,10 +123,10 @@ func (td TestData) externalProviders() map[string]resource.ExternalProvider {
 
 func PreCheck(t *testing.T) {
 	variables := []string{
-		"ARM_CLIENT_ID",
```

Comment:
Hi @matt-FFFFFF ,

I know this is still a draft but thought of letting you know that this is a breaking change. The `NewDefaultAzureCredential` function part of `azidentity` sdk uses env vars with `AZURE_*` as documented [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/README.md?plain=1#L170-L172). IMO `ARM_*` vars should still be supported as they are heavily used with `azurerm` provider.

Reply:
The use of `NewDefaultAzureCredential` means that you can test locally when logged in with az cli, or even use managed service identity in your self-hosted runner. Nonr of these methods require these environment variables to be set.

The removal of `ARM_CLIENT_ID` from the pre-check means that you do not _have_ to supply it but can still do so if you are using an authentication mechanism that requires them.

The go SDK will provide you a meaningful error if it cannot authenticate.

Reply:
okay, as I understood from you is that it's just internal testing of the provider. Nothing changes for the end user.